src/officehours_api/views.py

```diff
@@ -53,17 +53,34 @@ class UserList(DecoupledContextMixin, generics.ListAPIView):
     serializer_class = ShallowUserSerializer
 
 
-class UserDetail(DecoupledContextMixin, generics.RetrieveAPIView):
+class UserDetail(DecoupledContextMixin, LoggingMixin, generics.RetrieveUpdateAPIView):
     queryset = User.objects.all()
     permission_classes = (IsAuthenticated,)
 
-    def get(self, request, *args, **kwargs):
+    def get_serializer(self, instance, data=None, many=None, partial=None):
+        ctx = self.get_serializer_context()
+        kwargs = {}
+        if data:
+            kwargs['data'] = data
+        return (
+            MyUserSerializer(instance, context=ctx, **kwargs)
+            if instance == ctx['user']
+            else ShallowUserSerializer(instance, context=ctx, **kwargs)
+        )
+
+    def check_change_permission(self, request, user):
+        if user != request.user:
+            self.permission_denied(request)
```

Comment:
So you removed the `IsCurrentProfile` permission class, and this is kinda doing the work now of ensuring the current user is the user that's being changed. Would it make sense for this to be a `IsCurrentUser` permission class instead?

Reply:
If I applied it as a permission class, it would apply to the `Retrieve` method as well, but I want that to remain callable (and return a `ShallowUser`).

Reply:
Ah, that makes sense to me generally, even though I don't have a great handle on the permission classes.